packages/node-terra/src/indexer/apiterra.service.ts

```diff
@@ -233,3 +249,98 @@ export class TerraClient {
     return this.baseApi;
   }
 }
+
+export class TerraSafeApi implements ITerraSafeApi {
+  private _lcdConnection: AxiosInstance;
+  private _mantlemintConnection: AxiosInstance;
+  mantlemintHealthOK = false;
+
+  constructor(
+    private tendermintURL: string,
+    private height: number,
+    private readonly params?: Record<string, string>,
+    private mantlemintURL?: string,
+  ) {
+    const httpAgent = new http.Agent({ keepAlive: true, maxSockets: 10 });
+    const httpsAgent = new https.Agent({ keepAlive: true, maxSockets: 10 });
+
+    this._lcdConnection = axios.create({
+      httpAgent,
+      httpsAgent,
+      timeout: argv('node-timeout') as number,
+      baseURL: this.tendermintURL,
+      headers: {
+        Accept: 'application/json',
+        'User-Agent': `SubQuery-Node ${packageVersion}`,
+      },
+    });
+
+    if (this.mantlemintURL) {
+      this._mantlemintConnection = axios.create({
+        baseURL: this.mantlemintURL,
+        headers: {
+          Accept: 'application/json',
+          'User-Agent': `SubQuery-Node ${packageVersion}`,
+        },
+      });
+    }
+  }
```

Comment:
We want to reuse these connections, they should be shared between any instances of safeapi. This is to keep the connections limited to the endpoint